packages/asgardeo-ai/src/asgardeo_ai/agent_auth_manager.py

```
@@ -388,6 +508,41 @@ async def get_obo_token_with_ciba(
+        try:
+            switched_token = await self.token_client.get_token(
+                'organization_switch',
+                token=token,
+                switching_organization=switching_organization,
+                scope=scope_str
+            )
+            return switched_token
+            
+        except (TokenError, ValidationError):
+            raise
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)